packages/components/src/switch/src/SwitchField.tsx

```diff
@@ -24,7 +24,7 @@ const SwitchToDescriptionSizeAdapter: SizeAdapter<SwitchFieldProps["size"], Text
     md: "sm"
 };
 
-export interface SwitchFieldProps extends StyledSystemProps, BaseComponentProps {
+interface SwitchFieldRenderProps extends StyledSystemProps {
```

Comment:
I'm not so sure about this part.

If you look at Checkbox for example: https://github.com/adobe/react-spectrum/blob/68403fe55489dce3de1b3094c957d598ad719861/packages/react-aria-components/src/Checkbox.tsx#L59

They would have CheckboxProps and then the render props extends it like this RenderProps<CheckboxRenderProps>

and the render props is basically a bunch of booleans (may not always be the case)

![image](https://github.com/user-attachments/assets/50f8d4e1-411d-4dd4-8b7e-af767bb96a4b)

What I'm trying to say is CheckboxProps and RenderProps should be separate. I see you didn't use size in the useRenderProps, so it shouldn't be in the RenderProps.